app/services/CheckoutService.scala

```diff
@@ -328,22 +330,53 @@ class CheckoutService(identityService: IdentityService,
       )
+      val finalRenewCommand = applyPromoIfNecessary(contact, basicRenewCommand)
+      logger.info(s"Final renew command will be: ${finalRenewCommand}")
+      finalRenewCommand
+    }
 
-      val validPromotion = for {
+    def getValidPromotion(contact: Contact) = for {
         code <- renewal.promoCode.withLogging("promo code from client")
         deliveryCountryString <- contact.mailingCountry.withLogging("salesforce mailing country")
         deliveryCountry <- CountryGroup.countryByName(deliveryCountryString).withLogging("delivery country object")
         validPromotion <- promoService.validate[com.gu.memsub.promo.Renewal](code, deliveryCountry, renewal.plan.id).withLogging("validated promotion").toOption
-      } yield validPromotion
-      val withPromo = validPromotion.map(v => p.apply(v, prpId => catalog.paid.find(_.id == prpId).map(_.charges.billingPeriod).get, promoPlans)(renewCommand)).getOrElse(renewCommand)
-      zuoraService.renewSubscription(withPromo)
+    } yield validPromotion
+
+    def applyPromoIfNecessary(contact: Contact, basicRenewCommand: Renew): Renew = {
+      getValidPromotion(contact) match {
+        case Some(validPromotion) => {
+          logger.info(s"Attempting to apply the promotion: ${validPromotion.code} to subscription: ${subscription.name}")
+          p.apply(validPromotion, prpId => catalog.paid.find(_.id == prpId).map(_.charges.billingPeriod).get, promoPlans)(basicRenewCommand)
+        }
+        case None => {
+          logger.info(s"No promotion to apply, keeping basicRenewCommand")
+          basicRenewCommand
+        }
+      }
     }
 
     def ensureEmail(contact: Contact) = if (contact.email.isDefined) Future.successful(\/.right(()))
       else salesforceService.repo.addEmail(contact, renewal.email).map { either =>
         either.fold[Unit]({ err => logger.error(s"couldn't update email for sub: ${subscription.id}: $err") }, u => u)
       }
 
+    def getSubscriptionDetailsForEmail(maybeValidPromotion: Option[ValidPromotion[PromoContext]]): String = {
+      val currency = subscription.currency
+      val discountedPlanDescription = for {
+        validPromo <- maybeValidPromotion
+        discountPromotion <- validPromo.promotion.asDiscount
+      } yield {
+        renewal.plan.charges.prettyPricingForDiscountedPeriod(discountPromotion, currency)
+      }
+      def introductoryPeriodSubDescription = subscription.introductoryPeriodPlan.map { introductoryPlan =>
+        val nextRecurrringPeriod = subscription.recurringPlans.minBy(_.start)
+        introductoryPlan.charges.prettyPricing(currency) + " then " + nextRecurrringPeriod.charges.prettyPricing(currency)
+      }
+      val subscriptionDetails = discountedPlanDescription orElse introductoryPeriodSubDescription getOrElse renewal.plan.charges.prettyPricing(currency)
+      logger.info(s"Subscription details (for renewal email) will be: $subscriptionDetails")
+      subscriptionDetails
```

Comment:
could be subscriptionDetails.with[Context]Logging(s"Subscription details (for renewal email) will be: $subscriptionDetails")